packages/token-sdk/src/metadata/metadata-util.ts

```diff
@@ -0,0 +1,26 @@
+import { TokenMetadata } from "./types";
+
+export class MetadataUtil {
+  public static isPartial(metadata: Partial<TokenMetadata>): boolean {
+    return !metadata.name || !metadata.symbol || !metadata.image;
+  }
+
+  public static merge(...metadatas: Partial<TokenMetadata | null>[]): Partial<TokenMetadata> {
+    const merged: Partial<TokenMetadata> = {};
+    metadatas.forEach((metadata) => {
+      if (!metadata) {
+        return;
+      }
+      if (!merged.name) {
+        merged.name = metadata.name;
+      }
+      if (!merged.symbol) {
+        merged.symbol = metadata.symbol;
+      }
+      if (!merged.image) {
+        merged.image = metadata.image;
+      }
```

Comment:
spread op?
```
merged = {
   ...merged,
   ...metadata
}
```

Reply:
I think we need to explicitly check for existence of fields and overwrite. The spread operator will always overwrite to the last metadata, but the behavior we want is to preserve fields from previous metadatas if they were present.